### PR TITLE
Structural and editorial updates

### PR DESCRIPTION
**Pull request by domenic**
OK, we have a spec!! This PR just cleans it up a bit. It might look like a lot of changes but none of the actual interesting content was really changed. It's more just guidance on how to write these sort of things, in pull request form. I hope it's educational. Please review it to let me know if anything is wrong.

Commit message:

The main work here is clarifying the logger, formatter, and printer separation. They are now nested into their own section, and use function call notation to call each other. Printer was also factored out into its own section instead of being left vague.

Other changes are minor editorial ones, e.g. changing capitalization or phrasing, or adding argument lists to the method headings.

### CODE REVIEW

**Comment by terinjokes** [comment]
LGTM :+1:

File: index.bs [hunk 1]
```diff
@@ -41,58 +41,71 @@ Link Defaults: html (dfn) structured clone
 
 <h2 id="status" class="no-num no-toc">Status</h2>
 
-This specification is an early work in progress that welcomes feedback to refine toward more precise and compatible definitions.
+This specification is an early work in progress that welcomes feedback to refine toward more precise and compatible definitions. It is also the editors' first specification, so please be kind and constructive.
```
Comment (robertkowalski) [inline index.bs]: :heart:

**Comment by robertkowalski** [comment]
+1 thank you!

**Comment by robertkowalski** [comment]
merged as c55390b0243c5c68c54ed330136b025c22f5fcb4 with linebreaks after 72chars applied to commit message

**Pull request closed without merging**